libsession/src/main/java/org/session/libsession/snode/OnionRequestAPI.kt

```diff
@@ -190,8 +191,21 @@ object OnionRequestAPI {
                 if (unusedSnodes.count() < pathSnodeCount) { throw InsufficientSnodesException() }
                 // Don't test path snodes as this would reveal the user's IP to them
                 guardSnodes.minus(reusableGuardSnodes).map { guardSnode ->
-                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).map {
-                        val pathSnode = unusedSnodes.getRandomElement()
+                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).mapIndexed() { index, _ ->
+                        var pathSnode = unusedSnodes.getRandomElement()
+
+                        // For the last node: We need to make sure the version is >= 2.8.0
+                        // to help with an issue that will disappear once the nodes are all updated
+                        if(index == pathSize - 2) {
+                            val suitableSnodes = unusedSnodes.filter { Util.compareVersions(it.version, "2.8.0") >= 0 }
+                            pathSnode = if (suitableSnodes.isNotEmpty()) {
+                                suitableSnodes.random()
+                            } else {
+                                throw InsufficientSnodesException()
+                            }
+                        }
```

Comment:
Maybe something like this is (marginally more efficient and) resolves the `wtf` you get seeing the `pathSize - 2`

```suggestion
                    val result = buildList {
                        add(guardSnode)
                        addAll((1 until (pathSize)).mapIndexed { index ->
                            var pathSnode = unusedSnodes.getRandomElement()

                            // For the last node: We need to make sure the version is >= 2.8.0
                            // to help with an issue that will disappear once the nodes are all updated
                            if(index == pathSize - 1) {
                                val suitableSnodes = unusedSnodes.filter { Util.compareVersions(it.version, "2.8.0") >= 0 }
                                pathSnode = if (suitableSnodes.isNotEmpty()) {
                                    suitableSnodes.random()
                                } else {
                                    throw InsufficientSnodesException()
                                }
                            }
                        )
```

Reply:
I'm not too worried as this is to be removed on the next release.